tests/e2e/kafka_cluster_test.go

```diff
@@ -0,0 +1,72 @@
+// Copyright © 2023 Cisco Systems, Inc. and/or its affiliates
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/banzaicloud/koperator/api/v1beta1"
+	"github.com/gruntwork-io/terratest/modules/k8s"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+// createKafkaClusterIfDoesNotExist creates a kafka cluster if
+// there isn't a preexisting one
+func createKafkaClusterIfDoesNotExist(kubectlOptions *k8s.KubectlOptions, koperatorVersion string, sampleFile string) {
+	By("Checking existing kafkaClusters")
+	err := checkExistenceOfK8sResource(kubectlOptions, kafkaKind, kafkaClusterName)
+
+	if err == nil {
+		By(fmt.Sprintf("Kafka cluster %s already exists\n", kafkaClusterName))
+	} else {
+		By("Deploying a KafkaCluster")
+		requireApplyingKoperatorSampleResource(kubectlOptions, koperatorVersion, sampleFile)
+	}
+
+	return
+}
+
+// requireCreatingKafkaCluster creates a Kafka cluster and
+// checks the success of that operation.
+func requireCreatingKafkaCluster(kubectlOptions *k8s.KubectlOptions, koperatorVersion string, sampleFile string) {
+	When("Creating a Kafka cluster", func() {
+		It("Deploying a Kafka cluster", func() {
```

Comment:
If we want to follow Patrik's way (https://github.com/banzaicloud/koperator/blob/ef6db9416dc459e992587ad259c00da1ac2fa130/tests/e2e/koperator_test.go#L173) we should move the "It("Deploying a Kafka cluster", func()"
into requireCreateKafkaClusterIfDoesNotExist (https://github.com/banzaicloud/koperator/blob/ef6db9416dc459e992587ad259c00da1ac2fa130/tests/e2e/kafka_cluster_test.go#L31)